company-ycmd.el

```diff
@@ -222,16 +212,21 @@ the completion candidates.
 When `company-ycmd-enable-fuzzy-matching' is nil, check if
 candidate starts with PREFIX, whether to include candidate in
 candidates list."
-  (let ((completion-list (append completion-vector nil))
-        (candidates '()))
-    (dolist (candidate completion-list (nreverse candidates))
-      (when (or company-ycmd-enable-fuzzy-matching
-                (company-ycmd--prefix-candidate-p candidate prefix))
-        (let ((result (funcall construct-candidate-fn
-                               candidate prefix start-col)))
-          (if (listp result)
-              (setq candidates (append result candidates))
-            (push result candidates)))))))
+  (let* ((prefix-start-col (- (+ 1 (ycmd--column-in-bytes)) (length prefix)))
+         (prefix-size (- start-col prefix-start-col))
+         (prefix-str (substring-no-properties prefix 0 prefix-size))
```

Comment:
Calculate this only once per completion request, instead of for all candidates. It stays same for all candidates.